src/libcrun/cgroup.c

```diff
@@ -44,11 +44,11 @@ static const cgroups_subsystem_t cgroups_subsystems[] = {
   "devices",
   "pids",
   "memory",
-  "net_cls,net_prio",
```

Comment:
these two are usually mounted together.  Can you show the output for `grep cgroup /proc/self/mountinfo` ?

Reply:
root@f7c63618dde3:/opt/busybox# grep cgroup /proc/self/mountinfo
433 432 0:117 / /sys/fs/cgroup ro,nosuid,nodev,noexec shared:134 - tmpfs tmpfs ro,mode=755
434 433 0:30 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/cpuset rw,nosuid,nodev,noexec,relatime shared:135 master:19 - cgroup cpuset rw,cpuset
481 433 0:31 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/cpu rw,nosuid,nodev,noexec,relatime shared:136 master:20 - cgroup cpu rw,cpu
482 433 0:32 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/cpuacct rw,nosuid,nodev,noexec,relatime shared:137 master:21 - cgroup cpuacct rw,cpuacct
483 433 0:33 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/blkio rw,nosuid,nodev,noexec,relatime shared:138 master:22 - cgroup blkio rw,blkio
484 433 0:34 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime shared:139 master:23 - cgroup memory rw,memory
485 433 0:35 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/devices rw,nosuid,nodev,noexec,relatime shared:140 master:24 - cgroup devices rw,devices
486 433 0:36 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/freezer rw,nosuid,nodev,noexec,relatime shared:141 master:25 - cgroup freezer rw,freezer
487 433 0:37 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/net_cls rw,nosuid,nodev,noexec,relatime shared:142 master:26 - cgroup net_cls rw,net_cls
488 433 0:38 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/perf_event rw,nosuid,nodev,noexec,relatime shared:143 master:27 - cgroup perf_event rw,perf_event
489 433 0:39 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/net_prio rw,nosuid,nodev,noexec,relatime shared:144 master:28 - cgroup net_prio rw,net_prio
490 433 0:40 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/hugetlb rw,nosuid,nodev,noexec,relatime shared:145 master:29 - cgroup hugetlb rw,hugetlb
491 433 0:41 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/pids rw,nosuid,nodev,noexec,relatime shared:146 master:30 - cgroup pids rw,pids
492 433 0:42 / /sys/fs/cgroup/rdma rw,nosuid,nodev,noexec,relatime shared:147 master:31 - cgroup rdma rw,rdma
493 433 0:43 /docker/f7c63618dde3a5af6632fe0e7cc2d4e6be70be0b3f273d90b9f615e60f901c1e /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime shared:148 master:32 - cgroup cgroup rw,name=systemd
293 433 0:121 / /sys/fs/cgroup/unified rw,nosuid,nodev,noexec,relatime shared:149 - cgroup2 cgroup2 rw,nsdelegate

Reply:
Further more information, `runc` can handle this situation correctly because `runc` use the soft link short names to create cgroup dirs. I think it is no hurt to use the short names.

Reply:
thanks for the further information.  So the issue happens when running crun inside of another container?

I am fine with your fix, could you just show how cgroups look like with your patch applied?

Reply:
> So the issue happens when running crun inside of another container?

Yes, I use `runc` and `crun` in a container with `--privileged ` and with a `rw` mount of `/var/lib/docker`.

Reply:
> could you just show how cgroups look like with your patch applied?

```
root@f7c63618dde3:/opt/c/crun# ls /sys/fs/cgroup/net_cls/test
cgroup.clone_children  cgroup.procs  net_cls.classid  notify_on_release  tasks
root@f7c63618dde3:/opt/c/crun# ls /sys/fs/cgroup/net_prio/test
cgroup.clone_children  net_prio.ifpriomap  notify_on_release
cgroup.procs           net_prio.prioidx    tasks
root@f7c63618dde3:/opt/c/crun# ls /sys/fs/cgroup/cpu/test
cgroup.clone_children  cpu.cfs_period_us  cpu.rt_period_us   cpu.shares  notify_on_release
cgroup.procs           cpu.cfs_quota_us   cpu.rt_runtime_us  cpu.stat    tasks
root@f7c63618dde3:/opt/c/crun# ls /sys/fs/cgroup/cpuset/test
cgroup.clone_children  cpuset.mem_exclusive       cpuset.mems
cgroup.procs           cpuset.mem_hardwall        cpuset.sched_load_balance
cpuset.cpu_exclusive   cpuset.memory_migrate      cpuset.sched_relax_domain_level
cpuset.cpus            cpuset.memory_pressure     notify_on_release
cpuset.effective_cpus  cpuset.memory_spread_page  tasks
cpuset.effective_mems  cpuset.memory_spread_slab
```

Reply:
```
root@f7c63618dde3:/opt/c/crun# ls /sys/fs/cgroup/cpuacct/test
cgroup.clone_children  cpuacct.usage         cpuacct.usage_percpu_sys   cpuacct.usage_user
cgroup.procs           cpuacct.usage_all     cpuacct.usage_percpu_user  notify_on_release
cpuacct.stat           cpuacct.usage_percpu  cpuacct.usage_sys          tasks
```